CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -128,6 +136,14 @@ public static function add(&$params) {
       $financialAccount->opening_balance = $financialAccount->current_period_opening_balance = '0.00';
     }
     $financialAccount->save();
+
+    // invoke post hook
+    $op = 'create';
+    if (!empty($params['id'])) {
+      $op = 'edit';
+    }
```

Comment:
I think one can change the $params values in pre hook i.e they might unset($params['id']) when $op == 'edit' which will end up creating new entry in db rather update and hence post hook should invoke as $op == 'create'. 

 $op = empty($params['id']) ? 'create' : 'edit';
I was told by one of member from core team in past to use if () condition rather ternary operators. If this pattern is allowed i happy to optimize it and use it in future :)